core/node/gateway_migrator/src/lib.rs

```diff
@@ -56,19 +56,30 @@ impl GatewayMigrator {
                 tracing::info!("Stop signal received, GatewayMigrator is shutting down");
                 return Ok(());
             }
-            if self.settlement_layer
-                != current_settlement_layer(
-                    self.eth_client.as_ref(),
-                    gateway_client,
-                    &self.l1_settlement_layer_specific_contracts,
-                    self.l2_chain_id,
-                    &mut self.pool.connection().await?,
-                    &self.getters_facet_abi,
-                )
-                .await?
-            {
-                bail!("Settlement layer changed")
+            let current_settlement_layer = current_settlement_layer(
+                self.eth_client.as_ref(),
+                gateway_client,
+                &self.l1_settlement_layer_specific_contracts,
+                self.l2_chain_id,
+                &mut self.pool.connection().await?,
+                &self.getters_facet_abi,
+            )
+            .await;
+            match current_settlement_layer {
+                Ok(current_settlement_layer) => {
+                    if self.settlement_layer != current_settlement_layer {
+                        bail!("Settlement layer changed")
+                    }
+                }
+                Err(ContractCallError::EthereumGateway(err)) => {
+                    if !err.is_retriable() {
+                        tracing::error!("Failed to fetch data from sl {}", &err);
+                        return Err(err)?;
```

Comment:
`?` doesn't make sense here; you return an error anyway. Also, I'd refactor matching here to be more obvious:

```rust
match current_settlement_layer {
    Ok(current_settlement_layer) => { /* ... */ }
    Err(ContractCallError::EthereumGateway(err)) if err.is_retriable() => {
        tracing::info!("Transient error fetching data from SL: {err}");
    }
    Err(err) => {
        tracing::error!("Failed to fetch data from SL: {err}");
        return Err(err.into());
    }
}
```